### ISSUE TITLE
False positive for optional class attribute

### ISSUE THREAD

**Issue by CareF** [pylint-dev/pylint#9809]
### Bug description

```python
Example:


"""test_lint.py"""
from typing import Optional

class Foo:
    """class Foo"""
    a: Optional[dict] = None
    def empty(self):
        """foo"""

    @classmethod
    def init(cls):
        """init"""
        if cls.a is None:
            cls.a = {}
        cls.a[1] = 2
```

### Command used

```shell
pylint test_lint.py
```

### Pylint output

```python
************* Module test_lint
test_lint.py:15:8: E1137: 'cls.a' does not support item assignment (unsupported-assignment-operation)

------------------------------------------------------------------
Your code has been rated at 3.75/10 (previous run: 3.33/10, +0.42)
```

### Expected behavior

E1137 should not be triggered by `cls.a`

### Pylint version

```shell
pylint 3.2.5
astroid 3.2.2
Python 3.12.4 (main, Jun  6 2024, 18:26:44) [Clang 15.0.0 (clang-1500.3.9.4)]
```

### OS / Environment

MacOS with homebrew

### Additional dependencies

```python
pylint 3.2.5
astroid 3.2.2
Python 3.12.4 (main, Jun  6 2024, 18:26:44) [Clang 15.0.0 (clang-1500.3.9.4)]
```

**Comment by CareF** [comment]
duplicate with #9515

**Comment by CareF** [comment]
Adding assert does not help:

```python
"""module"""
from typing import Optional

class Foo:
    """class Foo"""
    a: Optional[dict] = None
    def empty(self):
        """foo"""

    @classmethod
    def init(cls):
        """init"""
        if cls.a is None:
            cls.a = {}
        assert cls.a is not None
        assert isinstance(cls.a, dict)
        if cls.a is None:
            return
        cls.a[1] = 2
```
